core/internal/testutils/testutils.go

```diff
@@ -380,6 +380,19 @@ func WaitForLogMessage(t *testing.T, observedLogs *observer.ObservedLogs, msg st
 	return
 }
 
+func WaitForLogMessageWithField(t *testing.T, observedLogs *observer.ObservedLogs, msg, field, value string) (le observer.LoggedEntry) {
+	AssertEventually(t, func() bool {
```

Comment:
`50ms` seems too fast to read all the logs. How about a longer period? And/Or would it be possible to `Take` the logs so that they are not reprocessed each time?

Reply:
Oh I see we have several pre-existing cases like this, so a separate ticket to follow up is probably fine.